tools/xsetwacom.c

```diff
@@ -2934,14 +2980,14 @@ int main (int argc, char **argv)
 	}
 
 	if (do_list)
-		list(dpy, argc - optind, &argv[optind]);
+		status = list(dpy, argc - optind, &argv[optind]);
 	else if (do_set)
-		set(dpy, argc - optind, &argv[optind]);
+		status = set(dpy, argc - optind, &argv[optind]);
 	else if (do_get)
-		get(dpy, format, argc - optind, &argv[optind]);
+		status = get(dpy, format, argc - optind, &argv[optind]);
 
 	XCloseDisplay(dpy);
-	return 0;
+	return status;
```

Comment:
```suggestion
	return status == 0 ? 0 : 1;
```

returning `-1` as an exit status is generally not ideal. the exit status is unsigned so -1 ends up being 255 in the shell and 128 and above usually indicate signal-based termination.

A better approach here would be to separate errors (exit status 1) from invalid usage (exit status 2) and return those values accordingly. `EXIT_SUCCESS` and `EXIT_FAILURE` exist in some system header file somewhere already, would be easy to `#define EXIT_INVALID_USAGE 2` and go with that then.

Note that invalid usage exit status 2 is merely one convention and not all programs use it.

Reply:
Thanks for the feedback, I'm not very well versed in the exit status meanings so I appreciate the explanation and suggestion I'll be sure to implement the suggested changes.

Reply:
Actually, found this one which explains it best: https://www.gnu.org/software/bash/manual/html_node/Exit-Status.html 

But for us, the only ones that matter are 0, 1 and 2

Reply:
> returning -1 as an exit status is generally not ideal. the exit status is unsigned so -1 ends up being 255

If the exit status is unsigned, should we claim the return value as "unsigned char"?

Reply:
that'd be breaking about 50 years of C convention :)

Reply:
Ha, that 50 years old specification left enough room for the return value to grow. That's quite robust ;).